app/models/concerns/account_counters.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+module AccountCounters
+  extend ActiveSupport::Concern
+
+  included do
+    has_one :account_stat, inverse_of: :account
+    after_save :save_account_stat
+  end
+
+  def statuses_count
+    account_stat&.statuses_count || 0
+  end
+
+  def statuses_count=(val)
+    (account_stat || build_account_stat).statuses_count = val
+  end
+
+  def following_count
+    account_stat&.following_count || 0
+  end
+
+  def following_count=(val)
+    (account_stat || build_account_stat).following_count = val
+  end
+
+  def followers_count
+    account_stat&.followers_count || 0
+  end
+
+  def followers_count=(val)
+    (account_stat || build_account_stat).followers_count = val
+  end
+
+  def increment_count!(key)
+    update_account_stat!(key => public_send(key) + 1)
```

Comment:
i think this should maybe be `attributes[key]`? or something similar, i can't remember the precise syntax offhand.

Reply:
That's if the counter was on the same table as before. We *want* to call the explicitly defined `followers_count` method here.